dotnet-authserver/tests/TeacherIdentity.AuthServer.Tests/AuthenticationStateTests.cs

```diff
@@ -224,11 +224,22 @@ public void GetInternalClaims_StaffUser_ReturnsExpectedClaims()
         AuthenticationState.AuthenticationMilestone expectedMilestone)
     {
         // Arrange
-        var linkGenerator = new Mock<IdentityLinkGenerator>(new Helpers.QueryStringSignatureHelper(key: "dummy"));
+        var linkGenerator = new Mock<LinkGenerator>();
+
+        var identityLinkGenerator = new TestIdentityLinkGenerator(
+            authenticationState,
+            linkGenerator.Object,
+            new Helpers.QueryStringSignatureHelper(key: "dummy"));
 
         void ConfigureMockForPage(string pageName, string returnsPath)
         {
-            linkGenerator.Protected().Setup<string>("Page", pageName, /* authenticationJourneyRequired: */ true)
+            linkGenerator
```

Comment:
This setup is a little gross but the entire `GetNextHopUrl()` method will be gone soon.